pkg/env/prod.go

```diff
@@ -320,20 +320,17 @@ func (p *prod) FeatureIsSet(f Feature) bool {
 	return p.features[f]
 }
 
-func (p *prod) FPAuthorizer(tenantID, resource string) (refreshable.Authorizer, error) {
-	oauthConfig, err := adal.NewOAuthConfig(p.Environment().ActiveDirectoryEndpoint, tenantID)
-	if err != nil {
-		return nil, err
-	}
-
+func (p *prod) FPAuthorizer(tenantID string, scopes ...string) (autorest.Authorizer, error) {
 	fpPrivateKey, fpCertificates := p.fpCertificateRefresher.GetCertificates()
 
-	sp, err := adal.NewServicePrincipalTokenFromCertificate(*oauthConfig, p.fpClientID, fpCertificates[0], fpPrivateKey, resource)
+	options := p.Environment().ClientCertificateCredentialOptions()
+	tokenCredential, err := azidentity.NewClientCertificateCredential(
+		tenantID, p.fpClientID, fpCertificates, fpPrivateKey, options)
```

Comment:
Same as above, we are shipping all certs instead of only the first.

Reply:
See above reply.